README.md

```diff
@@ -11,19 +11,42 @@ Currently the only final destination is Amplitude. We have plans to support more
 npm install --save analytics-client
 ```
 
-## Usage
 
-Tracking a page view. The following reports an event named `[EFP] Page View` with default properties.
+## Client Initialization
+The first thing to do is create and initialize an `analytics client`.
+
+Since most of balena products are maintained on different components/repos (ie Hub, Marketing Site, Cloud, etc) we need to make sure that
+when someone navigates between components we can keep identifying it as a same id. 
+
+So to achieve that we will need to:
 
 ```typescript
-import { createClient, createWebTracker } from 'analytics-client';
+import { AnalyticsUrlParams, createClient} from 'analytics-client';
+
+const urlParamsHandler = new AnalyticsUrlParams();
+
+// This will handle any passed device id (d_id) param on the URL if it exists, save it on a cookie and return a "clean" query string without the d_id param
+const newQuery = urlParamsHandler.consumeUrlParameters(window.location.search) ?? null;
+
+// Then if there was any passed device id we can obtain them like this.
+// passedDeviceId is either the first d_id param passed on the URL or null, so getPassedDeviceIds()[0] works even when no d_id is sent on the URL.
+const passedDeviceId = urlParamsHandler.getPassedDeviceIds()[0]
```

Comment:
Do we have any suspected usages of the multitude of passed device IDs?
At this point I expected a `getPassedDeviceId` (singular) method to handle only one ID as discussed in previous comments :)

Reply:
For sure it could happen. A user might come to Hub from various sources, marketing website, EFP, blog. Imagine someone using Etcher visiting the Hub for the first time via EFP and ending the session without further navigation or signup event. Then the same user later visits our blog from other means than EFP and navigates to Hub from the blog. On the Hub we will end up with multiple device identifiers to pass in the URL when the user navigates to the dashboard.

Reply:
On a related note, it'd be great to edit the Readme and this specific comment as we implement the new convenience method.

Reply:
> On the Hub we will end up with multiple device identifiers to pass in the URL when the user navigates to the dashboard.

I'd use `allDeviceIds` method for this instead of `getPassedDeviceIds`, but looking at the comment below I think you mean the same thing :)

Reply:
Oh now I get your first comment. Yes we are on the same page : )

Reply:
Right, there was no use case I could think about for all the passed ids, but I proceeded this way to differentiate between Ids that are passed from components from those that could be generated by the component in case it was needed afterwards for some reason.

thats why the method gets all the passed device ids and not only one. If we decide to build a function to get just one passed id, should it always be the first one from the array? Or could it be some other one for some reason? Since I wasnt sure, I though this function at least gives us the chance to decide afterwards.

But let me know and I could change it or create a new one for just the "last device id passed" (first one on the array)